R/utils.R

```diff
@@ -22,16 +22,26 @@ base_pkgs <- function() {
 #' @importFrom pkgcache ppm_snapshots
 get_ppm_snapshot_by_date <- function(date) {
   snaps <- pkgcache::ppm_snapshots()
-  res <- as.character(as.Date(head(
+  res <- as.character(as.Date(utils::head(
     snaps[as.Date(snaps$date) > as.Date(date), "date"],
     1
   )))
-  if (length(res) == 0) stop(sprintf("Cannot find PPM snapshot for date after %s.", as.character(date)))
+  if (length(res) == 0) {
+    rlang::warn(sprintf(
+        paste0(
+          "Cannot find PPM snapshot for date after %s.",
+          " Will use current CRAN instead."
+        ),
+        as.character(date)
+    ))
+    return(NA)
+  }
   res
 }
 
 #' @importFrom pkgcache ppm_repo_url
-parse_ppm_url <- function(snapshot) {
+parse_ppm_url <- function(snapshot = NA) {
+  if (is.na(snapshot)) return(pkgcache::default_cran_mirror())
```

Comment:
Fallback to current CRAN in case package published date is `NA`  _(this applies directly in case workaround is reverted)_